<launch>
    <arg name="spoof_spheros" default='true'/>

    <group if="$(eval spoof_spheros == 'true')">
        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="sf8_ekf">
            <param name="output_frame" value="odom_combined"/>
            <param name="base_footprint_frame" value="base_footprint"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>  
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="true"/>
            <param name="vo_used" value="false"/>
            <remap from="imu_data" to="sf8/imu_data" />
            <remap from="odom" to="sf8/odom" />
        </node>
        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="sd9_ekf">
            <param name="output_frame" value="odom_combined"/>
            <param name="base_footprint_frame" value="base_footprint"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>  
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="true"/>
            <param name="vo_used" value="false"/>
            <remap from="imu_data" to="sd9/imu_data" />
            <remap from="odom" to="sd9/odom" />
        </node>
    </group>

    <group if="$(eval spoof_spheros == 'false')">
        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="sca_ekf">
            <param name="output_frame" value="odom_combined"/>
            <param name="base_footprint_frame" value="base_footprint"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>  
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="false"/>
            <param name="vo_used" value="false"/>
            <remap from="imu_data" to="sca/imu_data" />
            <remap from="odom" to="sca/odom" />
        </node>

        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="sec_ekf">
            <param name="output_frame" value="odom_combined"/>
            <param name="base_footprint_frame" value="base_footprint"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>  
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="false"/>
            <param name="vo_used" value="false"/>
            <remap from="imu_data" to="sec/imu_data" />
            <remap from="odom" to="sec/odom" />
        </node>
    </group>

</launch>